From 42cedd1c88640e7f3e72cbd5877a84f66f393799 Mon Sep 17 00:00:00 2001
From: "Wang, ArvinX" <arvinx.wang@intel.com>
Date: Wed, 5 Jun 2019 14:54:07 +0800
Subject: [PATCH 3/3] Fix unable to recognize video files with ASF

The video files with ASF stored in the audio_playlists
table on Q. Moving it to the video table for keeping
the same with P.

Change-Id: Id05c03ce775c66eb3f31defd9f69a8f51dd7a6ac
Tracked-On: https://jira.devtools.intel.com/browse/OAM-81093
Signed-off-by: Wang, ArvinX <arvinx.wang@intel.com>
---
 src/com/android/providers/media/scan/ModernMediaScanner.java | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/com/android/providers/media/scan/ModernMediaScanner.java b/src/com/android/providers/media/scan/ModernMediaScanner.java
index d9e9859c..de03090f 100644
--- a/src/com/android/providers/media/scan/ModernMediaScanner.java
+++ b/src/com/android/providers/media/scan/ModernMediaScanner.java
@@ -492,12 +492,12 @@ public class ModernMediaScanner implements MediaScanner {

             if (attrs.isDirectory()) {
                 return scanItemDirectory(existingId, file, attrs, mimeType, volumeName);
-            } else if (MediaFile.isPlayListMimeType(mimeType)) {
-                return scanItemPlaylist(existingId, file, attrs, mimeType, volumeName);
             } else if (MediaFile.isAudioMimeType(mimeType)) {
                 return scanItemAudio(existingId, file, attrs, mimeType, volumeName);
             } else if (MediaFile.isVideoMimeType(mimeType)) {
                 return scanItemVideo(existingId, file, attrs, mimeType, volumeName);
+            } else if (MediaFile.isPlayListMimeType(mimeType)) {
+                return scanItemPlaylist(existingId, file, attrs, mimeType, volumeName);
             } else if (MediaFile.isImageMimeType(mimeType)) {
                 return scanItemImage(existingId, file, attrs, mimeType, volumeName);
             } else {
--
2.21.0
